openpype/pipeline/publish/lib.py

```diff
@@ -791,3 +791,22 @@ def _validate_transient_template(project_name, template_name, anatomy):
         raise ValueError(("There is not set \"folder\" template in \"{}\" anatomy"  # noqa
                              " for project \"{}\"."
                          ).format(template_name, project_name))
+
+
+def add_repre_files_for_cleanup(instance, repre):
+    """ Explicitly mark repre files to be deleted.
+
+    Should be used on intermediate files (eg. review, thumbnails) to be
+    explicitly deleted.
+    """
+    files = repre["files"]
+    staging_dir = repre.get("stagingDir")
```

Comment:
Yes, can definitely happen since it used to originally be stored on the `instance.data["stagingDir"]` and wasn't always set on the representation. I believe that hasn't been changed since and in some cases it's still used from the instance.

Reply:
> Can this actually happen that a representation is not having stagingDir?

I actually think that most of representations don't have it and are using staging dir from instance (not sure here...). I think that representation with staging dir is usually not pointing to same as staging dir on instance.